libraries/AP_HAL_Linux/Storage.cpp

```diff
@@ -20,26 +20,96 @@ using namespace Linux;
 
 // name the storage file after the sketch so you can use the same board
 // card for ArduCopter and ArduPlane
-#if APM_BUILD_TYPE(APM_BUILD_Replay)
-#define STORAGE_DIR "."
-#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RST_ZYNQ
-#define STORAGE_DIR "/mnt/APM"
-#else
-#define STORAGE_DIR HAL_BOARD_STORAGE_DIRECTORY
-#endif
-
-#define STORAGE_FILE STORAGE_DIR "/" SKETCHNAME ".stg"
+#define STORAGE_FILE SKETCHNAME ".stg"
 
 extern const AP_HAL::HAL& hal;
 
+static inline int is_dir(const char *path)
+{
+    struct stat st;
+
+    if (stat(path, &st) >= 0)
```

Comment:
miss {}

Reply:
I'm not sure if it worth changing. As said in the commit message this is copied code from kmod, something I did a long time ago.

Humn... Maybe it is. Will think.

Reply:
done